### PR TITLE
ci: add latest release workflows

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

Sub of #9

**Which issue(s) this PR fixes**:

- [ ] Latest Release Workflow:
  - [ ] Automatically create latest tag and release for using latest verson of i2gw.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Xunzhuo** [comment]
xref: https://github.com/kubernetes-sigs/ingress2gateway/issues/21

File: .github/workflows/latest-release.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          i2gw_latest_linux_amd64.tar.gz
+          i2gw_latest_linux_arm64.tar.gz
+          i2gw_latest_darwin_amd64.tar.gz
+          i2gw_latest_darwin_arm64.tar.gz
+        body: |
+          This is the "latest" release of **Ingress2Gateway**, which contains the most recent commits from the main branch.
+          
+          This release **might not be stable**.
+
+          It is only intended for developers wishing to try out the latest features in Ingress2Gateway, some of which may not be fully implemented.
```
Comment (robscott) [inline .github/workflows/latest-release.yaml:58]: This does not match Kubernetes/Gateway API versioning or release process. I think it's important for this project to be consistent with other Kubernetes projects so I'd rather leave the concept of a `latest-release` representing the latest commit on main out of this. Of course I think more general purpose release tooling would be helpful, just want it to be consistent with the rest of the Kubernetes org.

**Review by robscott — COMMENTED** [review]
Thanks @Xunzhuo for working on this! This is something that I'd rather see more closely match the Kubernetes release process.

**Comment by robscott** [comment]
@Xunzhuo let us know if you want to follow up on these, for now I'm going to close out in favor of more recent PRs.

**Pull request closed without merging**